src/System.Private.CoreLib/shared/System/Resources/ResourceReader.cs

```diff
@@ -743,7 +755,61 @@ private object _LoadObjectV2(int pos, out ResourceTypeCode typeCode)
             }
 
             // Normal serialized objects
-            throw new NotSupportedException(SR.NotSupported_ResourceObjectSerialization);
+            int typeIndex = typeCode - ResourceTypeCode.StartOfUserTypes;
+            return DeserializeObject(typeIndex);
+        }
+
+        private object DeserializeObject(int typeIndex)
+        {
+            if (!_permitDeserialization)
+            {
+                throw new NotSupportedException(SR.NotSupported_ResourceObjectSerialization);
+            }
+
+            if (_binaryFormatter == null)
+            {
+                InitializeBinaryFormatter();
+            }
+
+            Type type = FindType(typeIndex);
+  
+            object graph = s_deserializeMethod(_binaryFormatter, _store.BaseStream);
+            
+            // guard against corrupted resources
+            if (graph.GetType() != type)
+                throw new BadImageFormatException(SR.Format(SR.BadImageFormat_ResType_SerBlobMismatch, type.FullName, graph.GetType().FullName));
+ 
+            return graph;
+        }
+
+        private void InitializeBinaryFormatter()
+        {
+            LazyInitializer.EnsureInitialized(ref s_binaryFormatterType, () =>
+                Type.GetType("System.Runtime.Serialization.Formatters.Binary.BinaryFormatter, System.Runtime.Serialization.Formatters, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a",
```

Comment:
I suspect this will require rd.xml for .NET Native. I'm not sure if the tests are still broken though.

Reply:
Also, should we really just pull BinaryFormatter into CoreLib?

Reply:
No we should not pull BinaryFormatter into CoreLib.  WRT rd.xml we aren't adding that for any other of these lightup cases.

Reply:
Whether rd.xml is necessary depends on exactly how the code is written and what you're reflecting over. .NET Native should recognize the Type.GetType call and include the BinaryFormatter class, but since you're doing these calls in separate delegates, I don't think it'll be able to connect that to the GetMethod call, so it may not keep the Deserialize method around. It'd be good to test with it if possible.